src/main/java/com/redhat/devtools/lsp4ij/console/explorer/actions/ExportServerAction.java

```diff
@@ -0,0 +1,128 @@
+import com.redhat.devtools.lsp4ij.server.definition.launching.UserDefinedLanguageServerDefinition;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+/**
+ * Export one or more language servers to a zip file.
+ */
+public class ExportServerAction extends AnAction {
+    private final List<LanguageServerDefinition> languageServerDefinitions;
+    private static final Logger LOGGER = LoggerFactory.getLogger(ExportServerAction.class);
+
+    public ExportServerAction(List<LanguageServerDefinition> languageServerDefinitions) {
+        this.languageServerDefinitions = languageServerDefinitions;
+        if (this.languageServerDefinitions.size() == 1) {
+            getTemplatePresentation().setText(LanguageServerBundle.message("action.lsp.console.explorer.export.server.text"));
+            getTemplatePresentation().setDescription(LanguageServerBundle.message("action.lsp.console.explorer.export.server.description"));
+        } else {
+            getTemplatePresentation().setText(LanguageServerBundle.message("action.lsp.console.explorer.export.servers.text"));
+            getTemplatePresentation().setDescription(LanguageServerBundle.message("action.lsp.console.explorer.export.servers.description"));
+        }
+        getTemplatePresentation().setIcon(AllIcons.ToolbarDecorator.Export);
+    }
+
+    @Override
+    public void actionPerformed(@NotNull AnActionEvent e) {
+        FileChooserFactory fileChooserFactory = FileChooserFactory.getInstance();
+        FileSaverDescriptor fileSaverDescriptor = new FileSaverDescriptor(
+                LanguageServerBundle.message("action.lsp.console.explorer.export.servers.zip.save.title"), LanguageServerBundle.message("action.lsp.console.explorer.export.servers.zip.save.description"));
+        FileSaverDialog fileSaverDialog = fileChooserFactory.createSaveFileDialog(fileSaverDescriptor, e.getProject());
+        VirtualFileWrapper fileWrapper = fileSaverDialog.save("export.zip");
+        if (fileWrapper != null) {
+            VirtualFile virtualFile = fileWrapper.getVirtualFile(true);
+            if (virtualFile != null) {
+                ApplicationManager.getApplication().runWriteAction(() -> {
+                    try {
+                        virtualFile.setBinaryContent(createZipFromLanguageServers());
+                    } catch (IOException ex) {
+                        LOGGER.warn(ex.getLocalizedMessage(), e);
+                    }
+                });
+            }
+        }
+    }
+
+    /**
+     * Creates a zip file by handling each user defined language server definitions
+     * @return zip file as a byte array
+     */
+    private byte[] createZipFromLanguageServers() throws IOException {
```

Comment:
Instead of having this method in the action, I think we should have a clean API in the LanguageServerTemplateManager:

 * an `import` method which expect a zip file
 * an `export` method which expect a virtual file where the zip must be saved.

After that you could even write test for import and export by using LanguageServerTemplateManager